scripts/packaging/createupmpackages.ps1

```diff
@@ -0,0 +1,207 @@
+    "xrplugin" = "Assets\MRTK\Providers\XRSDK";
+    # extensions
+    "handphysicsservice" = "Assets\MRTK\Extensions\HandPhysicsService";
+    "losttrackingservice" = "Assets\MRTK\Extensions\LostTrackingService";
+    "scenetransitionservice" = "Assets\MRTK\Extensions\SceneTransitionService";
+    # other packages
+    "tools" = "Assets\MRTK\Tools";
+    "testutilties" = "Assets\MRTK\Tests\TestUtilities";
+    "examples" = "Assets\MRTK\Examples";
+}
+
+$npmPath = "npm"
+
+# Ensure we can call npm.cmd to package and publish
+[boolean]$nodejsInstalled = $false
+try {
+    node.exe -v > $null 2> $null
+    $nodejsInstalled = $true
+}
+catch {}
+
+if ($nodejsInstalled -eq $false)
+{
+    # Acquire and unpack node.js
+    $archiveName = "node-v$NodejsVersion-win-x64"
+    $archiveFile = ".\$archiveName.zip"
+    $downloadUri = "https://nodejs.org/dist/v$NodejsVersion/$archiveFile"
+    Write-Output "Downloading node.js v$NodejsVersion"
+    Invoke-WebRequest -Uri $downloadUri -OutFile $archiveFile
+    Write-Output "Extracting $archiveFile"
+    Expand-Archive -Path $archiveFile -DestinationPath ".\" -Force
+
+    $npmPath = "$scriptPath\$archiveName\$npmPath"
+}
+
+# Beginning of the upm packaging script main section
+# The overall structure of this script is:
+#
+# 1) Copy the appropriate publishing (.npmrc) file and set the npm command
+# 2) Replace the %version% token in the package.json file with the value of PackageVersion
+# 3) Overwrite the package.json file
+# 4) Create and publish the packages (local publishing copies the package to the OutputFolder)
+# 5) Cleanup files created and/or modified
+
+$npmrcFile = ".npmrc"
+$npmrcFileFullPath = "$projectRoot\scripts\packaging\$npmrcFile.$OutputTarget"
+$cmdFullPath = "$env:systemroot\system32\cmd.exe"
+$npmCommand = "pack"
+$updateAuth = $true
+$isLocalBuild = ($OutputTarget -eq "local")
+
+# Create and publish the packages
+foreach ($entry in $packages.GetEnumerator()) {
+    $packageFolder = $entry.Value
+    $packagePath = "$projectRoot\$packageFolder"
+    $npmrcBackup = "$npmrcFile.mrtk-bak"
+  
+    # Switch to the folder containing the package.json file
+    Set-Location $packagePath
+
+    # Backup any existing .npmrc file
+    if (Test-Path -Path $npmrcFile ) {
+        Rename-Item -Path $npmrcFile  -NewName $npmrcBackup
+    }
+
+     if (-not ($isLocalBuild)) {
+        # Copy the appropriate .nmprc file
+        Copy-Item -Path $npmrcFileFullPath -Destination ".\$npmrcFile" -Force
+
+        # Set the npm command to "publish"
+        $npmCommand = "publish"
+    }
+
+    # Get/update the credentials needed to access the server.
+    # This only needs to happen when credentials are updated. We run this script
+    # once per build to ensure the machine is ready to publish.
+    if (($updateAuth -eq $true) -and (Test-Path -Path $npmrcFile )) {
+        Start-Process -FilePath "$PSHOME\powershell.exe" -ArgumentList "vsts-npm-auth.ps1 -config $npmrcFile" -NoNewWindow -Wait

```

Comment:
Is this needed / will this work on the pipeline machines? I think we're able to just give the build service permissions to push to the feed

![image](https://user-images.githubusercontent.com/3580640/87100432-42d17300-c201-11ea-8c42-34449b633c58.png)

Reply:
i recall needing to do this step. worst case, it reports that it isnt needed :)